Genome and genome catalogues

MGS-sails opened the pull request:
This pull request involves the following:
1. The Migrations to create Genome, Genome Catalogues, and their related tables gotten from V1 APIS
2. Prefect flow to import Biomes
3. Prefect flow to import Genomes
4. APIs that expose genomes and genome-catalogues

MGS-sails committed:
created migrations for all related genomes table from V1 apis

MGS-sails committed:
created migrations for all related genomes table from V1 apis

MGS-sails committed:
Created biome import prefect flow

MGS-sails committed:
Created biome import prefect flow

MGS-sails committed:
Created Genome and catalogues api

MGS-sails committed:
Created Genome and catalogues api

SandyRogers commented on genomes/models/GenomeCatalogue.py:
```diff
@@ -0,0 +1,79 @@
+        help_text='The biome label for the catalogue (and any others that share the same practical biome). '
+                  'Need not be a GOLD biome, e.g. may include host species.'
+    )
+    PROK = 'prokaryotes'
+    EUKS = 'eukaryotes'
+    VIRS = 'viruses'
+    CATALOGUE_TYPE_CHOICES = (
+        (PROK, PROK),
+        (EUKS, EUKS),
+        (VIRS, VIRS),
+    )
```
This could probably now user helper `DjangoChoicesCompatibleStrEnum`

SandyRogers commented on genomes/models/GenomeCatalogue.py:
```diff
@@ -0,0 +1,79 @@
+
+    class Meta:
+        unique_together = ('catalogue_biome_label', 'version', 'catalogue_type')
+        db_table = 'genome_catalogue'
```
In general we aren't bothering to specify `db_table` in API V2 – just let django name them.

SandyRogers commented on genomes/models/GenomeCatalogue.py:
```diff
@@ -0,0 +1,79 @@
+    def __str__(self):
+        return self.name
+
+    def calculate_genome_count(self):
+        self.genome_count = self.genomes.count()
+        self.save()
```
See comment above re. using a GeneratedField perhaps

SandyRogers commented on workflows/tests/coverage/coverage.xml:
this one shouldn't be checked in / needs gitignore update

MGS-sails replied:
This has been removed

SandyRogers commented on workflows/tests/test_import_biomes_flow.py:
I think this is already covered okay by `test_importers.py`

MGS-sails replied:
The test has been removed

SandyRogers commented on workflows/flows/import_v5_biomes.py:
Cool – but any particular reason to wrap this management command in a prefect flow? I think for "one off" imports like this, a normal management command is kind of fine.

MGS-sails replied:
Not really required per say. But kept as a way to run the command in an easy when after this goes live. 
Can delete the flow afterwards

SandyRogers replied:
Ah cool – so this management command has already been run on the Prod DB so we won't need to re-run to release this.

SandyRogers commented on genomes/management/lib/genome_util.py:
Nice. I know these are mostly ported from API V1, but it would be great to refactor the file/dir sanity checkers here to use the APIV2-style validators, that is e.g. `DirectoryRule`, `GlobRule`, `FileRule` etc. I would guess copilot/cursor/junie etc would do a pretty good job of refactoring this using the helpers provided by `workflows/data_io_utils/file_rules/*`

SandyRogers reviewed (changes requested):
Thanks for this @MGS-sails ! Nice work getting to grips with the whole system.
Just a couple of suggestions to bring this PR a bit closer to the V2 patterns we've adopted:
1. Rather than all of the Downloads models, and associated base classes etc, I think we should use the `WithDownloadsModel` pattern (i.e. all of a genome's / catalogue's download objects are schematized json objects in a single jsonfield on the genome/catalogue db row). It cuts out loads of boilerplate and query JOINs.
2. Similarly with the annotations (COGs etc), I think adopting the pattern used for `annotations` field on the `Analysis` model would make a simpler data model.

I know both of these are a bit anti-relational... but I think they're reasonable choices and best to use them in Genomes as well, for consistency if nothing else.

You also mentioned needing transfer services support here – I think that can also copy the pattern used for Analysis, with a copier flow and an `external_results_dir` field on the models.

MGS-sails commented on docker-compose.yaml:
```diff
@@ -112,7 +112,7 @@ services:
     container_name: app
     profiles: ["app", "all"]
-    command: "runserver 0:8000"
+    command: "runserver 0:8009"
```
Dev debris. Will change it back

MGS-sails replied:
This has been removed

SandyRogers commented:
@MGS-sails one thing I forgotten in the review – a couple of test fixture files (e.g. the same ones as API V1, or some programticaly generated ones) would be useful here, to populate a couple of genomes into the dev db (i.e. included in `test_make_dev_data`). That will make it easier to work on the web client side of this feature.

MGS-sails committed:
WIP, added withDownloadsPattern to genome model

MGS-sails committed:
WIP, added withDownloadsPattern to genome model

dependabot[bot] committed:
Bump dj-database-url from 2.3.0 to 3.0.0

Bumps [dj-database-url](https://github.com/jazzband/dj-database-url) from 2.3.0 to 3.0.0.
- [Release notes](https://github.com/jazzband/dj-database-url/releases)
- [Changelog](https://github.com/jazzband/dj-database-url/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jazzband/dj-database-url/compare/v2.3.0...v3.0.0)

---
updated-dependencies:
- dependency-name: dj-database-url
  dependency-version: 3.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

SandyRogers committed:
adds jwt authentication for webin

SandyRogers committed:
prevent parallel test execution bludgeoning nf trace file

SandyRogers committed:
cleanup

SandyRogers committed:
hacky wait for prefect test flow to finish

SandyRogers committed:
adds private data storage and serving

SandyRogers committed:
Prefect / CI fixes (#68)

SandyRogers committed:
fix type hint

SandyRogers committed:
adds ignore file for jetbrains ai

SandyRogers committed:
adds support for private data in amplicon flow

SandyRogers committed:
improves test safety of private data location assertion

SandyRogers committed:
updates handling of LIBRARY_LAYOUT / FASTQs single/paired discrepancies

Ge94 committed:
Make old tests work

Ge94 committed:
Add checks for _2 alone and no fastqs

Ge94 committed:
Fix no fastqs use case

Ge94 committed:
Typo

Ge94 committed:
Linting

timrozday-mgnify committed:
Added raw reads files, functions, classes and configs throughout.

timrozday-mgnify committed:
Added raw reads files, functions, classes and configs throughout.

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

timrozday-mgnify committed:
Update pyproject.toml

Ge94 committed:
Small logic change

Ge94 committed:
Add logger warning

Co-authored-by: Ekaterina Sakharova <49755902+KateSakharova@users.noreply.github.com>

Ge94 committed:
Linting

MGS-sails committed:
Rewriting import command in light of With Downloads and Annotations pattern

MGS-sails committed:
Rewriting import command in light of With Downloads and Annotations pattern

MGS-sails committed:
WIP, added annotations for genomes

MGS-sails committed:
WIP, added annotations for genomes

MGS-sails committed:
Rewrote import_genome command to match json based structure of genomes model

MGS-sails committed:
Rewrote import_genome command to match json based structure of genomes model

MGS-sails committed:
Refactored import_genomes command

MGS-sails committed:
Refactored import_genomes command

MGS-sails committed:
Rewrote prefect flow

MGS-sails committed:
Rewrote prefect flow

MGS-sails committed:
WIP, debugging import genome prefect flow

MGS-sails committed:
WIP, debugging import genome prefect flow

MGS-sails committed:
WIP, resolving failing flow deployments

MGS-sails committed:
WIP, resolving failing flow deployments

MGS-sails committed:
Merged with latest changes from main

SandyRogers committed:
adjust app container's entrypoint for better pycharm integration

SandyRogers committed:
improve junie guidelines for tests

SandyRogers committed:
switch to django-ninja-extra API, using perms for private data

MGS-sails committed:
Completed debugging on import genomes prefect flow

MGS-sails committed:
Completed debugging on import genomes prefect flow

MGS-sails committed:
Removed unused comments

MGS-sails committed:
Removed unused comments

MGS-sails committed:
Fixed genome detail api

MGS-sails committed:
Fixed genome detail api

MGS-sails committed:
implemented genome annotations deferral and api

MGS-sails committed:
implemented genome annotations deferral and api

MGS-sails committed:
Fixed data format bug on genome detail endpoint

MGS-sails committed:
Fixed data format bug on genome detail endpoint

SandyRogers committed:
switch secure links for private data from MD5 to SHA256 HMAC

MGS-sails committed:
Various:
Fixed pre commit hook errors; Removed deprecated related models; wrote tests for import genomes

MGS-sails committed:
Various:
Fixed pre commit hook errors; Removed deprecated related models; wrote tests for import genomes

MGS-sails committed:
ReRan pre commit

MGS-sails committed:
ReRan pre commit

MGS-sails committed:
Removed unused test placeholders

MGS-sails committed:
Removed unused test placeholders

MGS-sails committed:
Wip, fixing failing tests

MGS-sails committed:
Wip, fixing failing tests

MGS-sails committed:
Corrected use of caps in column names for catalogue

MGS-sails committed:
Corrected use of caps in column names for catalogue

MGS-sails committed:
Debugging wrong genome_accession columbn bug in prefect flow

MGS-sails committed:
Debugging wrong genome_accession columbn bug in prefect flow

MGS-sails committed:
Fixed genome_accesion column issues, by cleaning default genome data

MGS-sails committed:
Fixed genome_accesion column issues, by cleaning default genome data

MGS-sails committed:
Made tests pass. Slight refactoring on tests

MGS-sails committed:
Made tests pass. Slight refactoring on tests

MGS-sails committed:
Added related fields to genome api response. Updated schemas to be inline with established patterns

MGS-sails committed:
Added related fields to genome api response. Updated schemas to be inline with established patterns

MGS-sails committed:
Crated tests for genomes app models

MGS-sails committed:
Crated tests for genomes app models

MGS-sails commented on emgapiv2/api/genomes.py:
```diff
@@ -0,0 +1,23 @@
+    return Genome.objects.select_related("biome", "catalogue").prefetch_related("pangenome_geographic_range")
+
+
+@router.get("/{accession}", response=GenomeDetail, summary="Get genome by accession", operation_id="get_genome")
```
This has been addressed. 
The relevant schemas have been updated and the API returned related models, similar to how it works on the Analyses app

MGS-sails committed:
Fixed omissions of genome-catalogues from API router

MGS-sails committed:
Fixed omissions of genome-catalogues from API router

MGS-sails committed:
Removed dev debris from debugging internal docker network issues

MGS-sails commented on emgapiv2/settings.py:
```diff
@@ -44,7 +44,7 @@ def show_toolbar(request):
 
 
-ALLOWED_HOSTS = ["apiv2-dev.mgnify.org", "localhost", "www.ebi.ac.uk"]
+ALLOWED_HOSTS = ["apiv2-dev.mgnify.org", "localhost", "www.ebi.ac.uk", "host.docker.internal:8004"]
```
Was used during debugging. 
Now removed

MGS-sails committed:
Removed coverage files

MGS-sails committed:
Removed coverage files

MGS-sails committed:
import biomes flow test not required, as it's covered by test_importers.py

MGS-sails committed:
import biomes flow test not required, as it's covered by test_importers.py

MGS-sails committed:
Added genome models to dev data generation

MGS-sails committed:
Added genome models to dev data generation

MGS-sails commented:
> @MGS-sails one thing I forgotten in the review – a couple of test fixture files (e.g. the same ones as API V1, or some programticaly generated ones) would be useful here, to populate a couple of genomes into the dev db (i.e. included in `test_make_dev_data`). That will make it easier to work on the web client side of this feature.



This has also been added

MGS-sails committed:
Added downloads to genome detail api response. Ran pre commit hooks

MGS-sails committed:
Added downloads to genome detail api response. Ran pre commit hooks

SandyRogers committed:
switch secure link query params back to longform names

SandyRogers committed:
add a few more flakey-test workarounds

MGS-sails committed:
Added back omitted django setup line

MGS-sails committed:
Added back omitted django setup line

SandyRogers reviewed (changes requested):
Thank you @MGS-sails ! Loads of great implementation here and its a big chunk of the API V2 migration.
I left quite a lot of comments sorry, but while it looks like a lot they are mostly minor things where we can simplify, and mostly deletions! 😁 Cheers!

timrozday-mgnify committed:
Adding raw-reads flow

timrozday-mgnify committed:
Adding raw-reads flow

timrozday-mgnify committed:
Add back deleted file

timrozday-mgnify committed:
Add back deleted file

MGS-sails commented on genomes/schemas/GenomeCatalogue.py:
```diff
@@ -0,0 +1,28 @@
+    ftp_url: str
+    pipeline_version_tag: str
+    catalogue_biome_label: str
+    catalogue_type: str
```
Thanks @SandyRogers are you referring to only the catalogue_type variable here ?
or are you saying we should have enums for ftp_url, pipeline_version_tab and catalogue_biome_label as well ?

MGS-sails committed:
Removed non required task decorators in genomes prefect flow; Adjusted tests to match new structure
; Adjusted flow options to be exploded

MGS-sails committed:
Removed non required task decorators in genomes prefect flow; Adjusted tests to match new structure
; Adjusted flow options to be exploded

MGS-sails commented on workflows/flows/import_genomes_flow.py:
```diff
@@ -0,0 +1,338 @@
+
+
+@flow(name="import_genomes_flow")
+def import_genomes_flow(options: dict):
```
Options have been exploded 💥 💥 💥

MGS-sails commented on workflows/flows/import_genomes_flow.py:
```diff
@@ -0,0 +1,338 @@
+from genomes.models import GenomeCatalogue, Genome, GeographicLocation
+
+
+@task
```
Done

MGS-sails committed:
Added GenomeCatalogueList schema

MGS-sails committed:
Added GenomeCatalogueList schema

MGS-sails commented on genomes/schemas/GenomeCatalogue.py:
```diff
@@ -0,0 +1,28 @@
+
+
+class GenomeCatalogueDetail(GenomeCatalogueBase):
+    biome_id: Optional[int]
```
A List schema has been added for GenomeCatalogue

MGS-sails commented:
Thanks for the quick review @SandyRogers .
I've responded to the feedback given.
Cheers !

MGS-sails committed:
Added Biome to GenomeBase schema

MGS-sails committed:
Added Biome to GenomeBase schema

mberacochea commented on genomes/management/lib/__init__.py:
```diff
@@ -0,0 +1,84 @@
+logger = logging.getLogger(__name__)
+
+
+class EMGBaseCommand(BaseCommand):
```
This is from the old days.. I'm not sure if this is the best way of composing cli commands. I like click a lot.. so maybe - https://github.com/django-commons/django-click to make it easier to build cli commands? @SandyRogers @MGS-sails

MGS-sails replied:
Thanks @mberacochea I checked out djclick, seems pretty nice to have. But I think we don't need to install it this time, as we're no longer using these management commands for now. Just the Prefect flow

mberacochea commented on workflows/flows/import_genomes_flow.py:
```diff
@@ -0,0 +1,355 @@
+
+
+@flow(name="import_genomes_flow")
+def import_genomes_flow(
```
Are we at risk of having an invalid db connection? The import process could take some time, maybe we should refresh the db connection every now and then?

MGS-sails replied:
@mberacochea I've added close_old_connections before genome db operations:

```python
   close_old_connections()
    genome, _ = Genome.objects.update_or_create(
        accession=accession, defaults=genome_data
    )
```

and also:

```python
    for genome_dir in genome_dirs:
        close_old_connections()
        genome_accession = process_genome_dir(catalogue, genome_dir)
        genome_accessions.append(genome_accession)
 ```

mberacochea commented on workflows/flows/import_genomes_flow.py:
```diff
@@ -0,0 +1,355 @@
+    if total != with_annot:
+        logger.warning("Some genomes are missing annotations!")
+    if total != with_files:
+        logger.warning("Some genomes are missing downloads!")
```
```suggestion
        logger.error("Some genomes are missing downloads!")
```

mberacochea reviewed (changes requested):
Excellent stuff @MGS-sails . I left some comments :)

SandyRogers commented on workflows/flows/import_v5_biomes.py:
```diff
@@ -0,0 +1,24 @@
+
+
+@task
+def run_import_biomes(url: str):
```
I've a feeling we don't actually need this flow though? The idea is that:
- one-off adminy stuff (like populating the biomes to match API v1) is a boring old django management command
- regular stuff (per dataset, or on some time cadence, or event-trigger based) is a prefect flow

dependabot[bot] committed:
Update django-ninja requirement from ~=1.4.2 to ~=1.4.3

Updates the requirements on [django-ninja](https://github.com/vitalik/django-ninja) to permit the latest version.
- [Release notes](https://github.com/vitalik/django-ninja/releases)
- [Commits](https://github.com/vitalik/django-ninja/compare/v1.4.2...v1.4.3)

---
updated-dependencies:
- dependency-name: django-ninja
  dependency-version: 1.4.3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump pandas from 2.2.3 to 2.3.0

Bumps [pandas](https://github.com/pandas-dev/pandas) from 2.2.3 to 2.3.0.
- [Release notes](https://github.com/pandas-dev/pandas/releases)
- [Commits](https://github.com/pandas-dev/pandas/compare/v2.2.3...v2.3.0)

---
updated-dependencies:
- dependency-name: pandas
  dependency-version: 2.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SandyRogers committed:
webin-cli: add debug logging and aspera option (#75)

mberacochea committed:
Bump the miassembler version to v3.0.0

mberacochea committed:
Modify the assembly_study flows for miassembler v3

I've added also the "contaminant_reference" in the assemblies metadata as this could be useful to query later.

All assemblies should be decontaminated of human data, that is done in the pipeline configuration. But't it's possible to specify a different human reference of phix reference (not likely but... you never know)

# Conflicts:
#	workflows/flows/assemble_study_tasks/make_samplesheets.py

MGS-sails committed:
Removed import biomes flow

MGS-sails committed:
Removed import biomes flow

MGS-sails committed:
Reverted genomes tables naming to django defaults

MGS-sails committed:
Reverted genomes tables naming to django defaults

MGS-sails committed:
Removed explicit database option from genomes flow

MGS-sails committed:
Removed explicit database option from genomes flow

MGS-sails committed:
Replaced cacased db options with protect

MGS-sails committed:
Replaced cacased db options with protect

MGS-sails commented on genomes/models/genome.py:
```diff
@@ -0,0 +1,199 @@
+        return self.first_created.isoformat() if self.first_created else None
+
+    class Meta:
+        db_table = "genome_genomes"
```
This is done

MGS-sails commented on genomes/models/genome_catalogue.py:
```diff
@@ -0,0 +1,87 @@
+        db_column="result_directory", max_length=100, null=True, blank=True
+    )
+    biome = models.ForeignKey(
+        Biome, db_column="biome_id", on_delete=models.CASCADE, null=True, blank=True
```
CASCADE operations have been replaced with PROTECT

MGS-sails committed:
Replaced logger warnings with errors when genomes are missing

MGS-sails committed:
Replaced logger warnings with errors when genomes are missing

MGS-sails committed:
Added catalogue_id as primary key for genome_catalogues

MGS-sails committed:
Added catalogue_id as primary key for genome_catalogues

mberacochea committed:
Update workflows/flows/assemble_study_tasks/assemble_samplesheets.py

Co-authored-by: Sandy Rogers <sandyr@ebi.ac.uk>

mberacochea committed:
Update workflows/flows/assemble_study_tasks/assemble_samplesheets.py

Co-authored-by: Sandy Rogers <sandyr@ebi.ac.uk>

mberacochea committed:
Implement suggestions from PR

mberacochea committed:
Include the centralized nextflow config file for miassembler

mberacochea committed:
Fix test_assembly_study_flow.py unit test

MGS-sails commented on workflows/flows/import_genomes_flow.py:
```diff
@@ -0,0 +1,355 @@
+
+    options = parse_options(options)
+    catalogue = get_catalogue(options)
+    finalize_catalogue(catalogue)
```
This function is no longer required and has been removed. As genomes count will now be a dynamic property

mberacochea committed:
Merge pull request #78 from EBI-Metagenomics/bugfix/miassembler-v3.0

Add support for miassembler v3.0.1

MGS-sails committed:
Made genome count dynamic on catalogue model; removed unecessary references to old field

MGS-sails committed:
Made genome count dynamic on catalogue model; removed unecessary references to old field

MGS-sails committed:
Added CatalogueType Typing to GenomeCatalogue schema

MGS-sails committed:
Added CatalogueType Typing to GenomeCatalogue schema

SandyRogers committed:
removes broker prefix from username in jwt token

SandyRogers committed:
typo fixes for priv data

mberacochea committed:
Suspend assembly_flow to edit samplesheet - idea

timrozday-mgnify committed:
Updated schemas and importing functions, and finished test_anlaysis_tawreads_study_flow.

timrozday-mgnify committed:
Updated schemas and importing functions, and finished test_anlaysis_tawreads_study_flow.

SandyRogers committed:
fix typo in nextflow run cmd for assembler

SandyRogers committed:
Merge pull request #80 from EBI-Metagenomics/bugfix/miassembler-config

fix typo in nextflow run cmd for assembler

KateSakharova committed:
add inferred library where defined

KateSakharova committed:
Merge pull request #81 from EBI-Metagenomics/fix/bug-inferred-library-strategy

add inferred library layout where defined

SandyRogers committed:
allow editing of assembly samplesheets BEFORE first pipeline attempt

SandyRogers committed:
fix tests for new samplesheet editing option

SandyRogers committed:
Merge pull request #79 from EBI-Metagenomics/feature/assembly-suspend-to-edit-samplesheet

Suspend assembly_flow to edit samplesheet - idea

SandyRogers committed:
fixes indentation problem with samplesheet edit list

dependabot[bot] committed:
Update requests requirement from ~=2.32.3 to ~=2.32.4

Updates the requirements on [requests](https://github.com/psf/requests) to permit the latest version.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.32.3...v2.32.4)

---
updated-dependencies:
- dependency-name: requests
  dependency-version: 2.32.4
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

SandyRogers committed:
admin improvements
- upgrades unfold-admin
- removes custom tabbed+paginated+inline
- fixes watchers listing on chromium browsers

SandyRogers committed:
adds missing (but no-consequence) migration

chrisAta committed:
add "WGA" as an allowed library_strategy for both METAGENOMIC and METATRANSCRIPTOMIC experiment types

chrisAta committed:
add unit test

chrisAta committed:
fix unit test

chrisAta committed:
fix typo

chrisAta committed:
add extra bits to unit tests that I missed

chrisAta committed:
fix linting

chrisAta committed:
fix assembled_runs bug

chrisAta committed:
Merge pull request #88 from EBI-Metagenomics/bugfix/add-wga-as-metagenomic

Add WGA as an allowed library strategy for METAGENOMIC and METATRANSCRIPTOMIC experiment types

KateSakharova committed:
fix for assembler selection

KateSakharova committed:
Added test

KateSakharova committed:
Merge pull request #90 from EBI-Metagenomics/fix/samplesheet-for-inferred-ll

fix for assembler selection

ochkalova committed:
Add fallback to ena data portal if metagenome data portal returns no data (#89)

* update ENAAPIRequest to loop throught list of data portals

add retry to get_study_readruns_from_ena

update ENAAPIRequest to loop throught list of data portals

* move data_portals setting to ENAConfig

* add test case for the study that is in ena portal only

* linter fixes

* fix test definition and update an assertion

* add serialization method to data_portals

* move ENAAvailabilityException to ENAAPIRequest.get() method

* remove unneccessary should_mock filter

Co-authored-by: Sandy Rogers <sandyr@ebi.ac.uk>

---------

Co-authored-by: Sandy Rogers <sandyr@ebi.ac.uk>

SandyRogers committed:
adds policy for including OTHER library_strat runs in assembly
- also updates queryset-to-samplesheet utils

timrozday-mgnify committed:
Ingest function profile read count and coverage

timrozday-mgnify committed:
Ingest function profile read count and coverage

timrozday-mgnify committed:
Merge branch 'EBI-Metagenomics:main' into main

SandyRogers committed:
renaming inferred_metadata -> metadata_preferring_inferred for clarity

SandyRogers committed:
documents various helpers in the codebase

SandyRogers committed:
if NO amplicon analysis runs complete, do not try (and fail) to run summary generators

MGS-sails committed:
Added db connection refreshing, to prevent stale db connection issues

MGS-sails committed:
Added db connection refreshing, to prevent stale db connection issues

MGS-sails committed:
Ran pre commit

MGS-sails committed:
Ran pre commit

SandyRogers reviewed (changes requested):
@MGS-sails many thanks, no new comments - just flagged a few from last time that are unresolved and still relevant. None of them are big!

MGS-sails committed:
Deleted unused genomes admin model

MGS-sails committed:
Deleted unused genomes admin model

MGS-sails committed:
Moved vars into GenomeConfig file,
Added Django admin stubs for Genome and related models

MGS-sails committed:
Moved vars into GenomeConfig file,
Added Django admin stubs for Genome and related models

MGS-sails committed:
Ran pre commit

MGS-sails committed:
Ran pre commit

SandyRogers committed:
removes nextflow run names from pipeline jobs

SandyRogers committed:
do not clean mi-assembler nextflow dirs before resuming

timrozday-mgnify committed:
Minor changes related to PR request comments

timrozday-mgnify committed:
Minor changes related to PR request comments

timrozday-mgnify committed:
Merge branch 'main' of github.com:timrozday-mgnify/emgapi-v2

timrozday-mgnify committed:
Merge branch 'EBI-Metagenomics:main' into main

timrozday-mgnify committed:
Added rawreads multiqc (study and sample) ingestion.

timrozday-mgnify committed:
Added rawreads multiqc (study and sample) ingestion.

KateSakharova committed:
Try with specified workdir

timrozday-mgnify committed:
Refactored repetative code for better readability

timrozday-mgnify committed:
Refactored repetative code for better readability

KateSakharova committed:
workdir should be workDir

timrozday-mgnify committed:
Bugfixes related to refactoring code to avoid code duplication

timrozday-mgnify committed:
Bugfixes related to refactoring code to avoid code duplication

KateSakharova committed:
change dash

KateSakharova committed:
Let it be -w

timrozday-mgnify committed:
Ran pre-commit linting etc.... Have allowed hex encoded strings in test_analysis_rawreads_study_flow.py

timrozday-mgnify committed:
Ran pre-commit linting etc.... Have allowed hex encoded strings in test_analysis_rawreads_study_flow.py

KateSakharova committed:
Merge pull request #97 from EBI-Metagenomics/fix/D

workdir should be -w

timrozday-mgnify committed:
FIx amplicon tests by splitting into one test per file.

timrozday-mgnify committed:
Resolve merge conflicts

timrozday-mgnify committed:
ran pre-commit post-commit

timrozday-mgnify committed:
Allow non-insdc for amplicon analysis, to help pass a test

timrozday-mgnify committed:
Proper handling of non_insdc setting for amplicon test

timrozday-mgnify committed:
ran pre-commit post-commit

MGS-sails committed:
Deprecated Genome location models

MGS-sails committed:
Deprecated Genome location models

MGS-sails committed:
Ran pre-commit

MGS-sails committed:
Ran pre-commit

timrozday-mgnify committed:
Change accession in amplicon test private study to avoid clash

timrozday-mgnify committed:
Fix missing import

timrozday-mgnify committed:
Fix missing import

timrozday-mgnify committed:
Merge pull request #92 from timrozday-mgnify/main

raw-reads-analysis-pipeline prefect orchestration

MGS-sails committed:
Adjusted tests to pass in light db connection refresh changes

MGS-sails committed:
Adjusted tests to pass in light db connection refresh changes

MGS-sails committed:
Added example to genomebase

MGS-sails committed:
Added example to genomebase

MGS-sails committed:
Replaced timestamps with base timestamped model references

MGS-sails committed:
Replaced timestamps with base timestamped model references

MGS-sails committed:
Linked Download Files with thier Indexes.

MGS-sails committed:
Linked Download Files with thier Indexes.

MGS-sails reviewed (commented):
Thanks @SandyRogers .
Please see the summary below, based on the changes we agreed on.


  🧬 Genome and Genome Catalogues Implementation

  This PR implements a comprehensive genome and genome catalogues system with the following key improvements:

  ✨ Features Added

  - Download Management: Linked download files with their indexes for better file organisation
  - API Endpoints: Complete genome and catalogue APIs with detailed responses including downloads and annotations
  - Data Models: New genome models with proper relationships, catalogues, and biome integration
  - Admin Interface: Django admin stubs for genome management

  🔧 Technical Improvements

  - Database: Replaced timestamps with base timestamped model references for consistency
  - Configuration: Consolidated variables into GenomeConfig file for better maintainability
  - Testing: Comprehensive test suite with adjusted database connection handling
  - Data Flow: Rewrote Prefect flows for genome import with JSON-based structure

  🧹 Cleanup & Refactoring

  - Deprecated old genome location models
  - Removed unused admin models and test placeholders
  - Fixed column naming issues and data format bugs

  📊 Data Management

  - Dynamic genome count on catalogue model
  - Proper primary key handling for genome_catalogues
  - Database connection refresh to prevent stale connections

timrozday-mgnify committed:
Add raw reads codon nextflow config file tp config.py

timrozday-mgnify committed:
Update run_rawreads_pipeline_via_samplesheet.py to use custom (codon) nextflow config

timrozday-mgnify committed:
Reformatted

timrozday-mgnify committed:
Chnge to dummy path

timrozday-mgnify committed:
Change dummy apth extension

timrozday-mgnify committed:
Merge pull request #102 from EBI-Metagenomics/rawreads-dev

Use rawreads codon nextflow config defined in emgapi-v2 config.py

SandyRogers committed:
merge genomes implementation with django-ninja-extra api changes

SandyRogers committed:
Merge remote-tracking branch 'origin/genome-and-genome-catalogues' into genome-and-genome-catalogues

# Conflicts:
#	emgapiv2/api/__init__.py
#	emgapiv2/api/genomes.py
#	emgapiv2/api/schema_utils.py
#	emgapiv2/test_make_dev_data.py
#	genomes/fixtures/genome/conftest.py
#	genomes/models/genome.py
#	genomes/schemas/GenomeBase.py
#	genomes/schemas/GenomeDetail.py
#	genomes/tests/test_models.py

SandyRogers committed:
lint

SandyRogers committed:
fix fixture
